doc/Learn/Tutorials/Http/HowTo-Refit.md

```diff
@@ -0,0 +1,219 @@
+
+* This interface will be used to generate the strongly-typed client. It will be used to make requests to the web service and deserialize responses.
+
+* For the purposes of this tutorial, we will create and register an interface for the Chuck Norris facts web API 
+
+* Start by defining an interface `IChuckNorrisEndpoint` with a method `Search()` that returns a `Task` of type `ApiResponse<ChuckNorrisData>`:
+    ```csharp
+    [Headers("Content-Type: application/json")]
+    public interface IChuckNorrisEndpoint
+    {
+        [Get("/jokes/search")]
+        Task<ApiResponse<ChuckNorrisData>> Search(CancellationToken ct, [AliasAs("query")] string searchTerm);
+    }
+    ```
+
+    > [!NOTE]
+    > A class named `ChuckNorrisData` will be defined in the subsequent sections of this guide. It will be used to deserialize the response from the API. 
+
+* The `Headers` attribute is used to specify the `Content-Type` header for the request. `Get` specifies the relative path for a `Search` request. Notice that the `searchTerm` parameter for the request is aliased as `query` using an `AliasAs` attribute.
+
+### 3. Register the endpoint
+
+* The `AddRefitClient<T>()` extension method is used to register the endpoint with the service collection.
+
+* This extension method can take a delegate as its argument, but the recommended way to configure a HTTP client is to specify a configuration section name. This allows you to configure the added HTTP client using the `appsettings.json` file. 
+
+* Add the Refit client to the service collection with the following code:
+
+    ```csharp
+    protected override void OnLaunched(LaunchActivatedEventArgs args)
+    {
+        var appBuilder = this.CreateBuilder(args)
+            .Configure(hostBuilder =>
+            {
+                hostBuilder.UseHttp((context, services) =>
+                    services.AddRefitClient<IChuckNorrisEndpoint>(context)
+                );
+            });
+    ...
+    ```
+
+* Define a configuration section in the `appsettings.json` file. While the default behavior is to use the platform-native HTTP handler, this can be configured. 
+
+    ```json
+    {
+        "ChuckNorrisEndpoint": {
+            "Url": "https://api.chucknorris.io/",
+            "UseNativeHandler": true
+        }
+    }
+    ```
+
+* The `Url` property specifies the base URL for the API.
+
+### 4. Generate a data model from the API response
+
+* `ApiResponse<T>` is used to deserialize the response from the API. It is defined in the `Refit` package.
+
+* The response from the API is a JSON object with a `total` property and `result` that contains an array of `ChuckNorrisData` objects. The `ChuckNorrisData` class will be defined in the `Models` namespace.
+
+* We will use [Hoppscotch](https://hoppscotch.io) to make a HTTP request and inspect the response. Open it, and make a `GET` request to the `/jokes/search` endpoint with the query parameter `query` set to `fight`.
```

Comment:
Image to make this clear?

Reply:
First time I've seen Hoppscotch - can we double check that we don't already have similar tools recommended elsewhere (eg Fiddler or Proxyman or Charles .... ). Following that, can we post on Collab and get a consensus from the team on what tool we want to reference.
Note: I'm not against Hoppscotch, just want to have some agreement before we promote tools via our docs